xml_converter/CMakeLists.txt

```diff
@@ -6,13 +6,22 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 # Name Output File
 set(TARGET_NAME xml_converter)
 
+# Add Protobuf
+find_package(Protobuf REQUIRED)
+include_directories(${Protobuf_INCLUDE_DIRS})
+# include_directories()
+EXECUTE_PROCESS(COMMAND ${PROTOBUF_PROTOC_EXECUTABLE} --proto_path=generators/proto_templates --cpp_out=src/ generators/proto_templates/waypoint.proto) 
```

Comment:
`EXECUTE_PROCESS` is run when `cmake` itself is called. The protobuf generation should probably be run when `make` is run instead of when `cmake` is run.

Additionally these paths do not function correctly when making an "out of source" build, which is generally recommended to do. An example of an out of source build is:

```bash
mkdir build
cd build
cmake ..
make
./my_executable
```

This means that when the build process is complete you can just run

```bash
rm -rf build
```
and clean up all the changes made by the build process. In our system things are a tad bit weird because we commit the code we generated from the generator we wrote into the source directory, but I think that is fine because the generator itself is a part of this codebase. Maybe in the future once this is strongly tested the generated files themselves will live outside the codebase and they will also be generated by the makefile just like the proto code.